internal/pkg/fleetapi/action_test.go

```diff
@@ -80,3 +82,56 @@ func mapRawMessageVal(m map[string]interface{}, key string) json.RawMessage {
 	}
 	return nil
 }
+
+func TestActionsUnmarshalJSON(t *testing.T) {
+	t.Run("ActionUpgrade no start time", func(t *testing.T) {
+		p := []byte(`[{"id":"testid","type":"UPGRADE","data":{"version":"1.2.3","source_uri":"http://example.com"}}]`)
+		a := &Actions{}
+		err := a.UnmarshalJSON(p)
+		require.Nil(t, err)
+		action, ok := (*a)[0].(*ActionUpgrade)
+		require.True(t, ok, "unable to cast action to specific type")
+		assert.Equal(t, "testid", action.ActionID)
+		assert.Equal(t, ActionTypeUpgrade, action.ActionType)
+		assert.Empty(t, action.ActionStartTime)
+		assert.Empty(t, action.ActionExpiration)
+		assert.Equal(t, "1.2.3", action.Version)
+		assert.Equal(t, "http://example.com", action.SourceURI)
+	})
+	t.Run("ActionUpgrade with start time", func(t *testing.T) {
+		p := []byte(`[{"id":"testid","type":"UPGRADE","start_time":"2022-01-02T12:00:00Z","expiration":"2022-01-02T13:00:00Z","data":{"version":"1.2.3","source_uri":"http://example.com"}}]`)
+		a := &Actions{}
+		err := a.UnmarshalJSON(p)
+		require.Nil(t, err)
+		action, ok := (*a)[0].(*ActionUpgrade)
+		require.True(t, ok, "unable to cast action to specific type")
+		assert.Equal(t, "testid", action.ActionID)
+		assert.Equal(t, ActionTypeUpgrade, action.ActionType)
+		assert.Equal(t, "2022-01-02T12:00:00Z", action.ActionStartTime)
+		assert.Equal(t, "2022-01-02T13:00:00Z", action.ActionExpiration)
+		assert.Equal(t, "1.2.3", action.Version)
+		assert.Equal(t, "http://example.com", action.SourceURI)
+	})
+	t.Run("ActionPolicyChange no start time", func(t *testing.T) {
+		p := []byte(`[{"id":"testid","type":"POLICY_CHANGE","data":{"policy":{"key":"value"}}}]`)
+		a := &Actions{}
+		err := a.UnmarshalJSON(p)
+		require.Nil(t, err)
+		action, ok := (*a)[0].(*ActionPolicyChange)
+		require.True(t, ok, "unable to cast action to specific type")
+		assert.Equal(t, "testid", action.ActionID)
+		assert.Equal(t, ActionTypePolicyChange, action.ActionType)
+		assert.NotNil(t, action.Policy)
+	})
+	t.Run("ActionPolicyChange with start time", func(t *testing.T) {
+		p := []byte(`[{"id":"testid","type":"POLICY_CHANGE","start_time":"2022-01-02T12:00:00Z","expiration":"2022-01-02T13:00:00Z","data":{"policy":{"key":"value"}}}]`)
+		a := &Actions{}
+		err := a.UnmarshalJSON(p)
+		require.Nil(t, err)
+		action, ok := (*a)[0].(*ActionPolicyChange)
+		require.True(t, ok, "unable to cast action to specific type")
+		assert.Equal(t, "testid", action.ActionID)
+		assert.Equal(t, ActionTypePolicyChange, action.ActionType)
+		assert.NotNil(t, action.Policy)
```

Comment:
[Suggestion]
Check the return of `StartTime` and `Expiration` to ensure the given `start_time` and `expiration` are ignored.

Reply:
turns out actions will have `expiration` times, however that is only used (by elastic agent) when searching ES for actions